.gitignore

```diff
@@ -0,0 +1,75 @@
+*.a
+*.bak
+*.db
+*.flash
+*.gcda
+*.gcno
+*.log
+*.o
+*.lo
+*.la
+*.ninja*
+*.opendb
+*.orig
+*.pcap
+*.pyc
+*.suo
+*.swn
+*.swo
+*.swp
+*.trs
+*.user
+*.bak
+*~
+.deps
+.dirstamp
+.DS_Store
+.local-version
+.libs
+.vagrant
+aclocal.m4
+autom4te.cache
+build
+CMakeCache.txt
+CMakeFiles
+cmake_install.cmake
+configure
+config.log
+config.status
+doc/Doxyfile
+doc/html
+etc/cmake/openthread-config-generic.h
+include/openthread-config.h
```

Comment:
Many of these files do not pertain to this repo. Can we clean this up?

Example: https://github.com/openthread/ot-nrf528xx/blob/main/.gitignore